backend/cognito/cognito.template.yml

```
@@ -90,6 +125,34 @@ Resources:
             Trigger: CustomMessage
             UserPool: !Ref UserPool
 
+  SecurityCodeTextMessageSender:
+    # checkov:skip=CKV_AWS_115: "Ensure that AWS Lambda function is configured for function-level concurrent execution limit"
+    # checkov:skip=CKV_AWS_117: "Ensure that AWS Lambda function is configured inside a VPC"
+    # checkov:skip=CKV_AWS_116: "Ensure that AWS Lambda function is configured for a Dead Letter Queue(DLQ)"
+    # checkov:skip=CKV_AWS_173: "Check encryption settings for Lambda environmental variable"
+    Type: AWS::Serverless::Function
+    Metadata: # Manage esbuild properties
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        Sourcemap: true
+        EntryPoints:
+          - security-code-text-message-sender.ts
+    Properties:
+      CodeUri: src/handlers/
+      Handler: security-code-text-message-sender.lambdaHandler
+      Runtime: nodejs18.x
+      Architectures:
+        - x86_64
+      MemorySize: 1024
+      Policies:
+        Statement:
+          - Sid: AllowSmsSenderToDecrypt
+            Effect: Allow
+            Action: kms:Decrypt
+            Resource: arn:aws:kms:eu-west-2:494650018671:key/375918d9-c680-44ff-8db2-14c1df0f10b8
```

```suggestion
          - Effect: Allow
            Action: kms:Decrypt
            Resource: !Ref arn:aws:kms:...:...:key/${<reference key here>}
```
Use pseudo params and use the key ARN in the policy

---

addressed